packages/hydrogen/src/analytics-manager/AnalyticsProvider.tsx

```diff
@@ -250,15 +253,25 @@ function register(key: string) {
 // This functions attempts to automatically determine if the user can be tracked if the
 // customer privacy API is available. If not, it will default to false.
 function shopifyCanTrack(): boolean {
-  if (
-    typeof window !== 'undefined' &&
-    typeof window?.Shopify === 'object' &&
-    typeof window?.Shopify?.customerPrivacy === 'object' &&
-    typeof window?.Shopify?.customerPrivacy?.userCanBeTracked === 'function'
-  ) {
-    return window.Shopify.customerPrivacy.userCanBeTracked();
+  try {
+    // eslint-disable-next-line no-undef
+    if (
+      typeof window !== 'undefined' &&
+      typeof window?.Shopify === 'object' &&
+      typeof window?.Shopify?.customerPrivacy === 'object' &&
+      typeof window?.Shopify?.customerPrivacy?.analyticsProcessingAllowed ===
+        'function'
+    ) {
```

Comment:
good call .. in fact, since the whole thing is in a try-catch, the if state is not needed

The try catch is needed because this function may be run at server side